### PR TITLE
change location of docker images to ghcr.io

### PR DESCRIPTION
**Pull request by flavorjones**
See context from #71 and #87 

This PR moves from dockerhub `larskanis` account to the github registry for the rake-compiler org. It keeps support for the `DOCKERHUB_USER` environment variable but also supports an env var with a more appropriate name, `DOCKER_REGISTRY`.

@larskanis Would love your feedback on this!

### CODE REVIEW

File: test/test_environment_variables.rb [hunk 1]
```diff
@@ -9,11 +9,11 @@
 class TestEnvironmentVariables
   module Common
     TEST_PLATFORM = ENV["TEST_PLATFORM"] || "x64-mingw-ucrt"
-    DOCKERHUB_USER = ENV['DOCKERHUB_USER'] || "larskanis"
+    DOCKER_REGISTRY = ENV['DOCKERHUB_USER'] || ENV['DOCKER_REGISTRY'] || "ghcr.io/rake-compiler"
```
Comment (larskanis) [inline test/test_environment_variables.rb:12]: I don't think we need to preserve `DOCKERHUB_USER`. So far, it was only used within the project, not by the gem.

**Review by larskanis — COMMENTED** [review]
I agree with you about the better name `DOCKER_REGISTRY`.

File: lib/rake_compiler_dock/starter.rb [hunk 2]
```diff
@@ -49,7 +49,7 @@ def exec(*args)
             platform_postfix = jrubyvm ? "" : "-#{platform}"
             ENV['RCD_IMAGE'] ||
                 ENV['RAKE_COMPILER_DOCK_IMAGE'] ||
-                "larskanis/rake-compiler-dock-#{rubyvm}#{platform_postfix}:#{IMAGE_VERSION}"
+                "ghcr.io/rake-compiler/rake-compiler-dock-#{rubyvm}#{platform_postfix}:#{IMAGE_VERSION}"
```
[the remark this answers is not in the document]
Comment (flavorjones) [reply lib/rake_compiler_dock/starter.rb:52]: Yes! will fix.

**Comment by flavorjones** [comment]
@larskanis I think I'd like to use a single container repository for all the images. Would you be open to changing the container name and tag format from:

```
#{REGISTRY}/rake-compiler-dock-#{RUBYVM}#{PLATFORM_POSTFIX}:#{VERSION}
```

to:

```
#{REGISTRY}/rake-compiler-dock-image:#{VERSION}-#{RUBYVM}#{PLATFORM_POSTFIX}
```

So, for example:

- `ghcr.io/ghcr.io/rake-compiler/rake-compiler-dock-image:1.3.0-mri-x86_64-linux`
- `ghcr.io/ghcr.io/rake-compiler/rake-compiler-dock-image:1.3.0-jruby`
- `ghcr.io/ghcr.io/rake-compiler/rake-compiler-dock-image:snapshot-mri-arm64-darwin`


The main advantage is that we could use a single image repository, `ghcr.io/rake-compiler/rake-compiler-dock-images` (rather than a separate repository per platform), which would simplify administration. This is also similar to what many other projects do: use the label as an indicator of both version and "flavor" (e.g., `ruby/ruby:3.2-alpine`).

This would also still work with the idea of shipping different host architectures, see #79.

**Comment by larskanis** [comment]
Yes, there are so many images now, that we should move the platform into the tag part. I already thought about it and when we touch the logic now, it's a good chance to change it now.

**Comment by flavorjones** [comment]
OK, I'm traveling a bit today but will get to this in the next day or so.

**Comment by flavorjones** [comment]
Running a Nokogiri pipeline using this repo and the snapshot images here: https://github.com/sparklemotion/nokogiri/pull/2756

**Pull request merged**